cmd/diff_utils.go

```diff
@@ -0,0 +1,1131 @@
+		// Simple value - show as single line
+		fmt.Fprintf(os.Stdout, "     → Array item: %s\n", formatValue(change.Value))
+	}
+}
+
+// unwrapParentheses removes leading '(' and trailing ')' from a string if both are present.
+func unwrapParentheses(s string) string {
+	// Check length before accessing indices to prevent panic
+	if len(s) >= 2 && s[0] == '(' && s[len(s)-1] == ')' {
+		return s[1 : len(s)-1]
+	}
+	return s
+}
+
+// printArrayChangeCompact handles compact array change display.
+func printArrayChangeCompact(change SchemaChange, path string) {
+	prefix := getPrefixForChangeType(change.Type)
+
+	// Handle modifications differently - show old → new
+	if isModificationWithValues(change) {
+		fmt.Fprintf(os.Stdout, "%s %s (%s → %s)\n", prefix, path,
+			formatCompactValue(change.OldFullValue), formatCompactValue(change.NewFullValue))
+		return
+	}
+
+	summary := getValueSummary(change.Value)
+	// For array changes, clarify it's an array item - unwrap any parentheses from summary
+	summaryContent := unwrapParentheses(summary)
+	fmt.Fprintf(os.Stdout, "%s %s (array item: %s)\n", prefix, path, summaryContent)
+}
+
+// printBasicChange handles non-array changes.
+// Routes basic change display to the appropriate formatter based on mode.
+func printBasicChange(change SchemaChange, path, header string, options DiffOptions) {
+	if options.Verbose {
+		printBasicChangeVerbose(change, path, header, options)
+	} else {
+		printBasicChangeCompact(change, path, header)
+	}
+}
+
+// printBasicChangeVerbose handles verbose basic change display.
+func printBasicChangeVerbose(change SchemaChange, path, header string, options DiffOptions) {
+	icon := getIconForChangeType(change.Type)
+	fmt.Fprintf(os.Stdout, "  %s %s\n", icon, path)
+	// Print additional details for additions
+	if header == newConfigHeader {
+		printValueProperties(change.Value, options)
+	}
+}
+
+// printBasicChangeCompact handles compact basic change display.
+func printBasicChangeCompact(change SchemaChange, path, header string) {
+	prefix := getPrefixForChangeType(change.Type)
+
+	// Compact mode: one line per change with summary
+	if header == newConfigHeader {
+		summary := getValueSummary(change.Value)
+		fmt.Fprintf(os.Stdout, "%s %s %s\n", prefix, path, summary)
+	} else {
+		// Removal
+		fmt.Fprintf(os.Stdout, "%s %s (removed)\n", prefix, path)
+	}
+}
+
+// printModifications handles modification display.
+func printModifications(modifications []SchemaChange, options DiffOptions, icon, prefix string) {
+	for _, change := range modifications {
+		path := formatPath(change.Path)
+
+		if options.Verbose {
+			fmt.Fprintf(os.Stdout, "  %s %s\n", icon, path)
+
+			// For array changes, show the full array values
+			if isArrayIndex(change.Path) && change.OldFullValue != nil && change.NewFullValue != nil {
+				fmt.Fprintf(os.Stdout, "     → Changed from: %s\n", formatValue(change.OldFullValue))
+				fmt.Fprintf(os.Stdout, "     → Changed to: %s\n", formatValue(change.NewFullValue))
```

Comment:
Why doesn't this use the printArray functions? Does this cover some case that those don't is there a difference in output?